platform/src/main/python/dlpx/virtualization/platform/import_validations.py

```diff
@@ -129,39 +127,43 @@ def check_upgrade_operations(plugin_module):
             if plugin_op_type != 'UpgradeOperations':
                 continue
 
-            warnings.extend(_check_upgrade_args(
-                plugin_attrib, plugin_module.expected_upgrade_args))
+            warnings.extend(
+                _check_upgrade_args(plugin_attrib,
+                                    plugin_module.expected_upgrade_args))
 
     return warnings
 
 
 def _check_upgrade_args(upgrade_operations, expected_upgrade_args):
     """
-    Does named argument validation of all functions in dictionaries by looping
-    first through all the attributes in the UpgradeOperations for this plugin.
+    This function does named argument validation of all migration functions by
+    first looping through each of the migration helpers (platform_migrations
+    and lua_migrations) then looping through all those attributes.
     Any attributes that are not dictionaries that map migration_id ->
     upgrade_function are skipped. We then loop through every key/value pair
     of each of the dictionaries and validate that the argument in the defined
     function has the expected name.
     """
     warnings = []
 
-    for attribute_name, attribute in vars(upgrade_operations).items():
-        if attribute_name not in expected_upgrade_args.keys():
-            # Skip if not in one of the operation dicts we store functions in.
-            continue
-        #
-        # If the attribute_name was in the expected upgrade dicts then we know
-        # it is a dict containing migration id -> upgrade function that we can
-        # iterate on.
-        #
-        for migration_id, migration_func in attribute.items():
-            actual = inspect.getargspec(migration_func).args
-            expected = expected_upgrade_args[attribute_name]
-            warnings.extend(
-                _check_args(method_name=migration_func.__name__,
-                            expected_args=expected,
-                            actual_args=actual))
+    for _, migration_helper in vars(upgrade_operations).items():
```

Comment:
Do we use the `_` for anything? If not, we should just loop through the values?

Reply:
oh yup this makes sense. I'l change it. when I wrote this I was just trying to convert what was already there, hadn't thought about not needing the key! Thanks.